h2o R package render script, closes PUBDEV-3689

jangorecki opened the pull request:
This PR brings shell script which allows to build h2o R package without extra dependencies like gradlew, or even java. This script is also more readable format than gradlew procedures so can help in documenting this process. So far it was tested on linux.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/h2oai/h2o-3/537)
<!-- Reviewable:end -->

navdeep-G commented:
I see it was only tested on Linux. What about Mac OS? Windows would be another story, though...

navdeep-G commented on h2o-r/h2o-package/render-h2o.sh:
```diff
@@ -0,0 +1,53 @@
+set -e
+
+### Render h2o R package from h2o-3 as a standalone R package h2o
+# script will download recent nightly h2o.jar so you should always rebase your h2o package to recent master
```
Technically a rebase is needed if they are using top of tree. However, a user could just take the jar and launch it, right?

jangorecki replied:
Yes, if we have `inst/java/h2o.jar` already in `h2o-package` we can use this. Rebase to latest and re-download jar ensure there are no breaking changes between h2o R package and h2o jar, something like strict version check.

navdeep-G replied:
Okay, sounds good.

navdeep-G commented on h2o-r/h2o-package/render-h2o.sh:
```diff
@@ -0,0 +1,53 @@
+
+mkdir -p man inst/java
+
+export JAR_BUILD_NUMBER=`curl -s http://h2o-release.s3.amazonaws.com/h2o/master/latest | head -1`
```
Maybe add some exception handling here just in case the link is corrupted.

jangorecki replied:
`set -e` on top of the script should handle that as expected, stop.

navdeep-G replied:
Ah, yep. Great.

ledell commented:
@jangorecki I see that `export H2O_VERSION="3.11.0"` is hardcoded.  Is the plan to update this variable on each major release?

jangorecki commented:
@ledell This is generally for developers who needs to rebuild R package, so shouldn't be problem to edit locally. It is mentioned in comment in header of shell script. If there is a way to extract that value from repo then it could be dynamic. Important to note that java H2O can be slightly out of version sync unless you checkout git to particular build number. Script is mostly useful if you are developing changes to R code and you are fine to use recent build number for defined H2O_VERSION.

navdeep-G commented:
@ledell These changes look good to me. I'll merge once tests are green.

navdeep-G commented:
@h2o-ops please test!

jangorecki commented:
This PR looks to be still functional.

```sh
cd h2o-r/h2o-package
./render-h2o.sh
```
```
Downloading recent h2o.jar build number: 4645
First time using roxygen2. Upgrading automatically...
Updating roxygen version in /home/jan/git/h2o-3/h2o-r/h2o-package/DESCRIPTION
Writing NAMESPACE
...
h2o R package content produced from templates. Check package using:
R CMD build .
R CMD check --run-dontrun --run-donttest --as-cran h2o_3.25.0.4645.tar.gz
```

note that `H2O_VERSION` env var in the script has to match H2O jar version downloaded from h2o-release.s3.amazonaws.com.

As a result I was able to build R h2o package without the need to even install gradlew etc.

```sh
R CMD build .
```
```
* checking for file ‘./DESCRIPTION’ ... OK
* preparing ‘h2o’:
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
Removed empty directory ‘h2o/docs’
Removed empty directory ‘h2o/vignettes’
* building ‘h2o_3.25.0.4645.tar.gz’
```

And run check

```sh
R CMD check --run-dontrun --run-donttest --as-cran h2o_3.25.0.4645.tar.gz
```
```
...
* checking CRAN incoming feasibility ... NOTE
Maintainer: ‘Erin LeDell <erin@h2o.ai>’

Version contains large components (3.25.0.4645)

Package has FOSS license, installs .class/.jar but has no 'java' directory.

Size of tarball: 122798568 bytes
...
* checking package dependencies ... NOTE
Package suggested but not available for checking: ‘slam’
...
* checking installed package size ... NOTE
  installed size is 126.2Mb
  sub-directories of 1Mb or more:
    java  124.3Mb
...
* checking examples ... NOTE
Examples with CPU or elapsed time > 5s
                           user system elapsed
h2o.dct                  10.681  0.064  25.581
h2o.target_encode_apply   2.313  0.124  20.064
h2o.target_encode_create  1.244  0.057  17.088
h2o.getAutoML             1.056  0.040  34.695
h2o.automl                0.833  0.043  34.662
h2o.glm                   0.755  0.048  18.962
plot.H2OModel             0.623  0.020   8.315
h2o.abs                   0.308  0.008   5.471
* checking PDF version of manual ... OK
* DONE

Status: 4 NOTEs
```

If there is no intent to merge this PR then it should be closed.
